packages/tasit-action/src/TasitAction.test.js

```diff
@@ -328,21 +328,66 @@ describe("TasitAction.Contract", () => {
       expect(txSubscription.subscribedEventNames()).to.be.empty;
     });
 
-    it("should emit error event when orphan/uncle block occurs", async () => {
-      txSubscription = simpleStorage.setValue("hello world");
-
+    // Note: Block reorganization is the situation where a client discovers a
+    //  new difficultywise-longest well-formed blockchain which excludes one or more blocks that
+    //  the client previously thought were part of the difficultywise-longest well-formed blockchain.
+    //  These excluded blocks become orphans.
+    it("should emit error event when block reorganization occurs - block excluded", async () => {
       const confirmationFn = sinon.fake();
       const errorFn = sinon.fake();
 
       const confirmationListener = message => {
         confirmationFn();
       };
 
+      const errorListener = message => {
+        const { error, eventName } = message;
+
+        // Note: This assertion will not fail the test case (UnhandledPromiseRejectionWarning)
+        expect(error.message).to.equal(
+          "Your message has been included in an uncle block."
+        );
+
+        // But asserting fake function, if that throws, test case will fail.
+        errorFn();
+      };
+
+      const snapshotId = await createSnapshot(provider);
+
+      txSubscription = simpleStorage.setValue("hello world");
+
       txSubscription.on("confirmation", confirmationListener);
 
+      txSubscription.on("error", errorListener);
+
+      await mineBlocks(provider, 10);
+
+      expect(confirmationFn.called).to.be.true;
+
+      await revertFromSnapshot(provider, snapshotId);
+
+      await mineBlocks(provider, 10);
+
+      txSubscription.off("confirmation");
+      txSubscription = undefined;
```

Comment:
This can  happen in a `beforeEach` or `afterEach`

Reply:
I've added this comment:
```javascript
      // Note: Transaction no longer exists
      // If it isn't unsed, afterEach hook will execute waitForNonceToUpdate forever
      txSubscription.off("confirmation");
      txSubscription = undefined;
```

What do you think?

Reply:
Yes, that's perfect!

Reply:
That's not in the code yet though, right?

Reply:
I've just committed.

Reply:
Resolving this